src/fetch_test.php

```diff
@@ -36,7 +46,32 @@
         );
         $answerStmt = $conn->prepare("INSERT INTO answers (value, question_id, is_correct) VALUES (?, ?, ?)");
 
+        // Skip questions that are not due to criteria
         while (($data = fgetcsv($handle, 1000, ",")) !== FALSE) {
+            if ($creator && $data[1] !== $creator) {
+                continue;
+            }
+            if ($testPurpose && $data[3] !== $testPurpose) {
+                continue;
+            }
+            break;
```

Comment:
Not sure if this loop actually does something

Reply:
It repeatedly searches the CSV file until it locates a match for the popup's requirements. It looks for a question with the creator {creator} and purpose {purpose}, and then it begins looping over the list of questions.